examples/modbus/MBus_WTH_CO2_LCD_ETH/dds_writer/CMakeLists.txt

```
@@ -13,24 +13,13 @@
 #                                                                             #
 ###############################################################################
 
-include(${RESOURCE_DIR}/cmake/dds_functions.cmake)
-include(${RESOURCE_DIR}/cmake/common_functions.cmake)
-
-# Copy the USER_QOS_PROFILES before generating the code
-# List of the files to copy
-set(CO2_LCD_ETH_EXAMPLE_WRITER_QOS_FILE
-    "USER_QOS_PROFILES.xml"
-)
-
-copy_files(
-    CUSTOM_TARGET_NAME CO2_LCD_ETH_EXAMPLE_WRITER
-    OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}"
-    LIST_OF_FILES_RELATIVE_PATH ${CO2_LCD_ETH_EXAMPLE_WRITER_QOS_FILE})
+include("${RESOURCE_DIR}/cmake/DdsFunctions.cmake")
+include("${RESOURCE_DIR}/cmake/CommonFunctions.cmake")
 
 set(IDL_NAME "MBus_WTH_CO2_LCD_ETH_WRITE")
```

Pass name directly to `rtigw_connextdds_run_codegen()`. In general, I  think it would make things more readable caching the directory into a variable, since that's the longest common stem between various calls to `rtigw_connextdds_run_codegen()`
```suggestion
set(IDL_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../idl")
```

---

do you mean this directory? `${CMAKE_CURRENT_SOURCE_DIR}/../idl/`?

---

ok I have just seen the following comment in which you specify this